src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -288,7 +305,7 @@ public static ModelMetadata fromString(String modelMetadataString) {
         // Training node assignment was added as a field in Version 2.12.0
         // Because models can be created on older versions and the cluster can be upgraded after,
         // we need to accept model metadata arrays both with and without the training node assignment.
-        if (modelMetadataArray.length == 7) {
+        if (modelMetadataArray.length == 8) {
```

Comment:
Are we handling BWC when there are 7? @ryanbogan can you help review?

Reply:
Yeah we should either make this a switch or add a new if statement.  I think we need to always accept 7

Reply:
Thanks Jack and Ryan for the callout, let me update.